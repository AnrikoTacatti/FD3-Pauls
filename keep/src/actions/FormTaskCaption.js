

import api from '../api/api.js';
import { OPEN_FORM_NEW_TASK_CAPTION, OPEN_FORM_EDIT_TASK_CAPTION } from '../stores/const.js';



export const actioncloseFormNewTaskCaption = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_NEW_TASK_CAPTION, data: data });
}

export const actioncloseFormEditTaskCaption = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_EDIT_TASK_CAPTION, data: data });
}

export const actionsetNewTaskCaption = (data, dispatch) => {
    api.setNewTaskCaption(data, dispatch);
}

export const actioneditTaskCaption = (data, dispatch) => {
    api.editTaskCaption(data, dispatch);
}

