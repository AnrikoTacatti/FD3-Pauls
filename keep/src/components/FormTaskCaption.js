import React from 'react';
import { connect } from 'react-redux';
import {
    actioncloseFormNewTaskCaption,
    actioncloseFormEditTaskCaption,
    actionsetNewTaskCaption,
    actioneditTaskCaption
} from '../actions/FormTaskCaption.js';



export class FormTaskCaption extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            /* TaskLists: this.props.stateTaskLists.TaskLists*/
            openFormNewTaskCaption: false,
            openFormEditTaskCaption: false
        }
    }

    UNSAFE_componentWillReceiveProps(newProps) {
        this.setState({ openFormNewTaskCaption: newProps.openFormNewTaskCaption });
        this.setState({ openFormEditTaskCaption: newProps.openFormEditTaskCaption });
    }

    closeFormNewTaskCaption = () => {
        let data = false;
        actioncloseFormNewTaskCaption(data, this.props.dispatch)
    }

    closeFormEditTaskCaption = () => {
        let data = { active: false };
        actioncloseFormEditTaskCaption(data, this.props.dispatch);
    }

    addNewTaskCaption = () => {
        let data = { title: this.inputTaskListName.value };
        actionsetNewTaskCaption(data, this.props.dispatch);
    }

    editTaskCaption = () => {
        let data = { title: this.inputTaskListName.value, keychapter: this.state.openFormEditTaskCaption.keychapter };
        actioneditTaskCaption(data, this.props.dispatch);
    }


    render() {
        // console.log("FormTaskCaption", this.props);

        return (
            <React.Fragment>
                {this.state.openFormEditTaskCaption.active === true && <React.Fragment>
                    <div className="form-add-task-caption form">
                        <div className="form-inner">
                            <p className="form-title">Edit task list</p>
                            <input type="text" className="task-list-name" name="title" defaultValue={this.state.openFormEditTaskCaption.name} placeholder="Enter task list name" required="" ref={c => this.inputTaskListName = c} />
                        </div>
                        <div className="form-add-task-caption__footer">
                            <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormEditTaskCaption}>Cancel</button>
                            <button type="button" className="form-submit btn" onClick={this.editTaskCaption} >Submit</button>
                        </div>
                    </div>
                    <div className="form-bg"></div>
                </React.Fragment>
                }
                {this.state.openFormNewTaskCaption === true && <React.Fragment>
                    <div className="form-add-task-caption form">
                        <div className="form-inner">
                            <p className="form-title">Add task list</p>
                            <input type="text" className="task-list-name" name="phone" placeholder="Enter task list name" required="" ref={c => this.inputTaskListName = c} />
                        </div>
                        <div className="form-add-task-caption__footer">
                            <button type="button" title="Cancel" className="form-cancel btn" onClick={this.closeFormNewTaskCaption}>Cancel</button>
                            <button type="button" className="form-submit btn" onClick={this.addNewTaskCaption} >Submit</button>
                        </div>
                    </div>
                    <div className="form-bg"></div>
                </React.Fragment>
                }
            </React.Fragment>
        );

    }

}

const mapStateToProps = function (state) {
    return {
        // весь раздел Redux state под именем counters будет доступен
        // данному компоненту как this.props.counters
        openFormNewTaskCaption: state.stateTaskLists.openFormNewTaskCaption,
        openFormEditTaskCaption: state.stateTaskLists.openFormEditTaskCaption,
    };
};

export default connect(mapStateToProps)(FormTaskCaption);






