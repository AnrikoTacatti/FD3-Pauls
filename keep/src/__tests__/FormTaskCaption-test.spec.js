
"use strict";
import React from 'react';
import { FormTaskCaption } from '../components/FormTaskCaption.js';
jest.mock('../actions/FormTaskCaption.js');
import {
    actioncloseFormNewTaskCaption,
    actioncloseFormEditTaskCaption,
    actionsetNewTaskCaption,
    actioneditTaskCaption
} from '../actions/FormTaskCaption.js';


describe('FormTaskItem New', () => {
    let dispatch = jest.fn();
    actionsetNewTaskCaption.mockImplementation((dispatch, data) => { return data; });
    actioncloseFormNewTaskCaption.mockImplementation((dispatch, data) => { return data; });
    let UNSAFE_componentWillReceiveProps = jest.spyOn(FormTaskCaption.prototype, "UNSAFE_componentWillReceiveProps");

    const component = shallow(
        <FormTaskCaption dispatch={dispatch} />
    );
    component.setProps({
        openFormNewTaskCaption: true,
        openFormEditTaskCaption: false
    });

    test('ormTaskCaption New Snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('FormTaskCaption New UNSAFE_componentWillReceiveProps', () => {
        component.setProps({
            openFormNewTaskCaption: true,
            openFormEditTaskCaption: false
        });
        expect(UNSAFE_componentWillReceiveProps).toHaveBeenCalled();
    });

    test('FormTaskCaption New  form-title', () => {
        const title = component.find(".form-title").text();
        expect(title).toBe("Add task list");
    });

    test('FormTaskCaption New cancels', () => {
        const btn = component.find(".form-cancel");
        btn.simulate('click');
        expect(actioncloseFormNewTaskCaption).toHaveBeenCalledWith(false, dispatch);
    });

});


describe('FormTaskItem Edit', () => {

    let dispatch = jest.fn();
    actioneditTaskCaption.mockImplementation((dispatch, data) => { return data; });
    actioncloseFormEditTaskCaption.mockImplementation((dispatch, data) => { return data; });
    let UNSAFE_componentWillReceiveProps = jest.spyOn(FormTaskCaption.prototype, "UNSAFE_componentWillReceiveProps");

    const component = shallow(
        <FormTaskCaption dispatch={dispatch} />
    );
    component.setProps({
        openFormNewTaskCaption: false,
        openFormEditTaskCaption: { active: true, keychapter: "-MnosmgmdFWwUg_Qyj5o-test", name: "Заметки учеба" }
    });

    test('ormTaskCaption Edit Snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('FormTaskCaption Edit UNSAFE_componentWillReceiveProps', () => {
        component.setProps({
            openFormNewTaskCaption: false,
            openFormEditTaskCaption: { active: true, keychapter: "-MnosmgmdFWwUg_Qyj5o-test", name: "Заметки учеба" }
        });
        expect(UNSAFE_componentWillReceiveProps).toHaveBeenCalled();
    });

    test('FormTaskCaption Edit  form-title', () => {
        const title = component.find(".form-title").text();
        expect(title).toBe("Edit task list");
    });

    test('FormTaskCaption Edit cancels', () => {
        const btn = component.find(".form-cancel");
        btn.simulate('click');
        expect(actioncloseFormEditTaskCaption).toHaveBeenCalledWith({ active: false }, dispatch);
    });



});