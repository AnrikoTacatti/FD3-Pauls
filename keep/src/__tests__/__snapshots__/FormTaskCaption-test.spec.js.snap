// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTaskItem Edit ormTaskCaption Edit Snapshot 1`] = `
<Fragment>
  <div
    className="form-add-task-caption form"
  >
    <div
      className="form-inner"
    >
      <p
        className="form-title"
      >
        Edit task list
      </p>
      <input
        className="task-list-name"
        defaultValue="Заметки учеба"
        name="title"
        placeholder="Enter task list name"
        required=""
        type="text"
      />
    </div>
    <div
      className="form-add-task-caption__footer"
    >
      <button
        className="form-cancel btn"
        onClick={[Function]}
        title="Cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        className="form-submit btn"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
  <div
    className="form-bg"
  />
</Fragment>
`;

exports[`FormTaskItem New ormTaskCaption New Snapshot 1`] = `
<Fragment>
  <div
    className="form-add-task-caption form"
  >
    <div
      className="form-inner"
    >
      <p
        className="form-title"
      >
        Add task list
      </p>
      <input
        className="task-list-name"
        name="phone"
        placeholder="Enter task list name"
        required=""
        type="text"
      />
    </div>
    <div
      className="form-add-task-caption__footer"
    >
      <button
        className="form-cancel btn"
        onClick={[Function]}
        title="Cancel"
        type="button"
      >
        Cancel
      </button>
      <button
        className="form-submit btn"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
  <div
    className="form-bg"
  />
</Fragment>
`;
